### PR TITLE
YES 66 Not pulling configurable product data wrong product url resolved.

### PR DESCRIPTION
**Pull request by ijazalise**
YES 66 Not pulling configurable product data wrong product url resolved.

### CODE REVIEW

File: Helper/ApiClient.php [hunk 1]
```diff
@@ -29,56 +29,99 @@ public function __construct(\Magento\Store\Model\StoreManagerInterface $storeMan
...
+            $description = $this->_escaper->escapeHtml(strip_tags($rawdescription));
+            $product_data['description'] = $description;
+            if ($item->getData('product_type') == 'grouped') {
+                if (!isset($productPrice[$productID])) {
+                    $productPrice[$productID] = 0.0000;
+                }
+                $productPrice[$productID] += $item->getData('row_total_incl_tax');
+                $product_data['price'] = $productPrice[$productID];
+            } else {
+                if (!isset($productPrice[$productID])) {
```
Comment (DanGlz) [inline Helper/ApiClient.php]: The else statement is identical to the if statement, in both cases the same code executed.

File: Helper/ApiClient.php [hunk 2]
```diff
@@ -29,56 +29,99 @@ public function __construct(\Magento\Store\Model\StoreManagerInterface $storeMan
...
-				}
-				$rawdescription =  str_replace(array('\'', '"'), '', $full_product->getDescription()); 
-				$description =  $this->_escaper->escapeHtml(strip_tags($rawdescription));
-				$product_data['description'] = $description;
-				$product_data['price'] = $item->getPrice();
-				$products_arr[$full_product->getId()] = $product_data;
-			}
+            try {
+                $productID = $item->getProduct()->getId();
+                if ($item->getData('product_type') == 'simple') {
```
Comment (DanGlz) [inline Helper/ApiClient.php]: 1. get product type once in variable at the begining , product_type = $item->getData('product_type')

File: Helper/ApiClient.php [hunk 3]
```diff
@@ -29,56 +29,99 @@ public function __construct(\Magento\Store\Model\StoreManagerInterface $storeMan
...
+                    $_product = $productModel->load($productID);
+                }
+                if ($item->getData('product_type') == 'configurable' || $item->getData('product_type') == 'grouped' || $item->getData('product_type') == 'bundle') {
```
Comment (DanGlz) [inline Helper/ApiClient.php]: better to use else statement instead

File: Helper/ApiClient.php [hunk 4]
```diff
@@ -29,56 +29,99 @@ public function __construct(\Magento\Store\Model\StoreManagerInterface $storeMan
...
+                    if ($item->getData('product_type') == 'grouped') {
+                        $productIDs = $groupedProductModel->getParentIdsByChild($item->getProduct()->getId()) ? $groupedProductModel->getParentIdsByChild($item->getProduct()->getId()) : 0;
```
Comment (DanGlz) [inline Helper/ApiClient.php]: use this variable '$groupedProductModel->getParentIdsByChild($item->getProduct()->getId())' once the the beginning of the scope

File: Helper/ApiClient.php [hunk 5]
```diff
@@ -29,56 +29,99 @@ public function __construct(\Magento\Store\Model\StoreManagerInterface $storeMan
...
+                        $productID = $productIDs[0];
+                    }
+                    if ($productCollection->create()->count()) {
+                        $productCollection->create()->clear();
+                        $_product = $item->getProduct();
+                    }
+                    $_products = $productCollection->create()->addAttributeToSelect('*')->addStoreFilter()->addFieldToFilter('entity_id', ['in' => $productID]);
+                    foreach ($_products as $product) { // This is needed as we can't use object as array in collection way.
```
Comment (DanGlz) [inline Helper/ApiClient.php]: why we can't use $_products[0] for example?

Comment (ijazalise) [reply Helper/ApiClient.php]: this would not work because it's associated array.

**Pull request merged**